src/main/scala/com/fulcrumgenomics/bam/SamRecordClipper.scala

```diff
@@ -248,6 +248,30 @@ class SamRecordClipper(val mode: ClippingMode, val autoClipAttributes: Boolean)
     else clipEndOfRead(rec, numberOfBasesToClip)
   }
 
+  /** Converts all soft-clipping to hard-clipping.  Does not convert unless the clipping mode is [[ClippingMode.Hard]].
+    * Calculates any clipping required and delegates to [[clipStartOfRead()]] and [[clipEndOfRead()]].
+    * @param rec the record to be clipped
+    * @return the number of bases converted to hard-clipping from soft-clipping at the start and end of the read
+    *         respectively.
+    */
+  def toHardClip(rec: SamRecord): (Int, Int) = {
+    if (rec.unmapped || this.mode != ClippingMode.Hard) (0, 0) else {
+      val numBasesClippedStart = {
+        val clippingElems = rec.cigar.elems.takeWhile(_.operator.isClipping)
+        val numSoft       = clippingElems.filter(_.operator == Op.S).map(_.length).sum
+        if (numSoft > 0) this.clipStartOfRead(rec, clippingElems.map(_.length).sum)
```

Comment:
I think I would just call directly to `upgradeClipping(rec, clippingElems.map(_.length).sum, fromStart=true)`.  I think that's a bit more transparent.